### PR TITLE
feature3.0

### PR DESCRIPTION
**Pull request by Liza101111**

### CODE REVIEW

File: src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class TaskController {
+
+    @Autowired
+    private TaskRepository taskRepository;
+
+    @PostMapping("/save")
```
Comment (veronycca) [inline src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java]: /task

File: src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public ResponseEntity<Task> saveTask(@RequestBody Task task){
+        Task savedTask = taskRepository.save(task);
+        return new ResponseEntity<>(savedTask, HttpStatus.OK);
+    }
+
+    @GetMapping("/tasks")
```
Comment (veronycca) [inline src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java]: /task

/task/{id}

Comment (veronycca) [reply src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java]: 1. add folder , model, -> TaskDto.

File: src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public ResponseEntity<List<Task>> listAllTasks(){
+        List<Task> tasks = taskRepository.findAll();
```
Comment (veronycca) [inline src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java]: for (Task task: tasks) {
TaskDto dto = new TaskDto();
dto.setName(task.getName());
}

File: src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    @GetMapping
+    public ResponseEntity<List<TaskDto>> getAllTasks(){
+
+        return ResponseEntity.ok().body(taskService.findAll());
+    }
+
+    @GetMapping("/{taskId}")
+    public ResponseEntity<TaskDto> getTaskById(@PathVariable Long taskId) {
+        TaskDto taskDto = taskService.findById(taskId);
+        return ResponseEntity.ok(taskDto);
```
Comment (veronycca) [inline src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java:36]: why do you use 
ResponseEntity.ok(body) here but ResponseEntity.ok().body(body) in a method above?
do you know a difference?

File: src/main/java/com/exercise/tasksmanagementsystem/service/TaskService.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+//        return taskRepository.findAll()
+//                .stream()
+//                .map(taskMapper::toDto)
+//                .collect(Collectors.toList());
+    }
+    public TaskDto findById(Long taskId){
+        Task task = taskRepository.findById(taskId)
+                .orElseThrow(()-> new ResourceNotFoundException("Task not found with id: " + taskId));
+        return taskMapper.toDto(task);
+    }
```
Comment (veronycca) [inline src/main/java/com/exercise/tasksmanagementsystem/service/TaskService.java:40]: use an empty line before each new method

File: src/main/java/com/exercise/tasksmanagementsystem/mapper/TaskMapper.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    public List<TaskDto> toDtoList(List<Task> tasks){
+        List<TaskDto> taskDtoList = new ArrayList<>();
+        for(Task task: tasks){
+            taskDtoList.add(toDto(task));
+        }
+        return taskDtoList;
+    }
+
+    public List<Task> toEntityList(List<TaskDto> taskDtos) {
```
Comment (veronycca) [inline src/main/java/com/exercise/tasksmanagementsystem/mapper/TaskMapper.java]: where do you use this method?

File: src/main/java/com/exercise/tasksmanagementsystem/service/TaskService.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+@Service
+public class TaskService {
+    @Autowired
+    private TaskRepository taskRepository;
+    @Autowired
+    private TaskMapper taskMapper;
```
Comment (veronycca) [inline src/main/java/com/exercise/tasksmanagementsystem/service/TaskService.java:19]: add an empty line here

File: src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java (hunk 4 above)
Comment (Liza101111) [reply src/main/java/com/exercise/tasksmanagementsystem/controller/TaskController.java:36]: Yes, I have read about their differences. ResponseEntity.ok(taskDto) and ResponseEntity.ok().body(taskDto) will return the same result. I just use these two methods to understand their usage. Maybe for more readable coding style, ResponseEntity.ok(taskDto) is better?

File: src/main/java/com/exercise/tasksmanagementsystem/mapper/TaskMapper.java (hunk 6 above)
Comment (Liza101111) [reply src/main/java/com/exercise/tasksmanagementsystem/mapper/TaskMapper.java]: I did not use this method, just to learn how to map EntityList and DtoList, so I leave this method there:)

**Pull request merged**